diff --git a/augene-player/Source/MainComponent.cpp b/augene-player/Source/MainComponent.cpp
index 4462956..ad80a2f 100644
--- a/augene-player/Source/MainComponent.cpp
+++ b/augene-player/Source/MainComponent.cpp
@@ -156,6 +156,7 @@ MainComponent::MainComponent()
                                                        engine.getPluginManager().knownPluginList,
                                                        engine.getTemporaryFileManager().getTempFile ("PluginScanDeadMansPedal"),
                                                        tracktion_engine::getApplicationSettings());
+        v->setHeadlessScanning(true);
         activePluginListComponent.reset(v);
         nextFormatForPseudoHeadlessScanning = 0;
         startTimer(100);
